Lib/pathlib.py

```diff
@@ -1166,6 +1189,210 @@ def walk(self, top_down=True, on_error=None, follow_symlinks=False):
 
             paths += [path._make_child_relpath(d) for d in reversed(dirnames)]
 
+    def absolute(self):
+        """Return an absolute version of this path by prepending the current
+        working directory. No normalization or symlink resolution is performed.
+
+        Use resolve() to get the canonical path to a file.
+        """
+        raise UnsupportedOperation(f"{type(self).__name__}.absolute()")
+
+    @classmethod
+    def cwd(cls):
+        """Return a new path pointing to the current working directory."""
+        return cls().absolute()
+
+    def expanduser(self):
+        """ Return a new path with expanded ~ and ~user constructs
+        (as returned by os.path.expanduser)
+        """
+        raise UnsupportedOperation(f"{type(self).__name__}.expanduser()")
+
+    @classmethod
+    def home(cls):
+        """Return a new path pointing to the user's home directory (as
+        returned by os.path.expanduser('~')).
+        """
+        return cls("~").expanduser()
+
+    def readlink(self):
+        """
+        Return the path to which the symbolic link points.
+        """
+        raise UnsupportedOperation(f"{type(self).__name__}.readlink()")
+
+    def resolve(self, strict=False):
+        """
+        Resolve '..' segments in the path. Where possible, make the path
+        absolute and resolve symlinks on the way.
+        """
```

Comment:
This implementation of `_PathBase.resolve()` is brand new, and probably the most complicated and interesting part of the patch.

It implements '..' segment and symlink resolution using POSIX rules, like `realpath --physical`. This makes it slightly incompatible with Windows rules.

It suppresses `UnsupportedOperation` exceptions from `absolute()`, which means returned paths may not be absolute. This is important for formats that support symlinks but not working directories (e.g. `.tar` archives).

Reply:
@eryksun would you have time to review this? It's an implementation of `_PurePath.resolve()` (~`os.path.realpath()`) that calls `p.absolute()` and `p.readlink()` under the hood, without performing any filesystem access of its own. This could be used by subclasses of `_PurePath` that need symlink support, like a `TarPath` class for tarballs.